core-common/src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionImpl.java

```diff
@@ -68,6 +68,9 @@ public ReturnType getData() throws NakadiRuntimeException {
     @Override
     public void close() {
         listener = null;
+        if (null != curatorFramework && !curatorFramework.isZk34CompatibilityMode()) {
```

Comment:
I am not sure I understand why `curatorFramework.isZk34CompatibilityMode()` used.
the doc says:
```
If mode is true, create a ZooKeeper 3.4.x compatible client. IMPORTANT: If the client
library used is ZooKeeper 3.4.x <code>zk34CompatibilityMode</code> is enabled by default.
```
I do not think we create curator framework instance in compatible mode and we use 3.6 zookeeper.

Reply:
It is safer to check it explicitly, because the `remove()` call on the next line will throw an `IllegalStateException` if in compatibility mode.

If somebody starts curator in this mode (and it's as easy as reverting the zookeeper dependency version: no actual _code_ changes), this will start throwing exceptions.

Reply:
in this case tests will show it, but we can also remove it once migrated to CF5.0.
thanks

Reply:
Yes, but the trouble is that this is only caught by the acceptance test, and the problem is exposed indirectly: quite a number of tests fail with unexpected results and the root cause needs to be figured out from the server logs.

I see the following options:
- Have a test that checks this condition explicitly and remove it from production code.
- Make AT failure reason more apparent (see also next option).
- Check it at server start and refuse to run in compatibility mode altogether.

Reply:
I am fine to have this check, because we will eventually remove it due to migration to curator framework >= 5.x, which does not have such method.

Reply:
@adyach `!= null` check should be removed (as it spoils all the attempts to use `@Nullable` annotations. Field is not nullable, therefore the check is not needed, and adds confusion

Reply:
I still do not understand who is the author of PR @a1exsh or @adyach

Reply:
It was mostly me.  Does it matter? :)

Reply:
Regarding `!= null` - matters, regarding author - it's just hard to follow the discussion

Reply:
Actually, it depends how you count.  The one liner code change was suggested by Andrey, and then we were trying to make it pass the tests ;-)